tests/core/snapd-refresh-vs-services-reboots/task.yaml

```diff
@@ -0,0 +1,140 @@
+  # revert to the previous revision of the snapd snap, and then we manually 
+  # remove all disabled revisions of the snapd snap - this manual removal will
+  # in fact unmount the mount units for /snap/snapd/x2 for example, avoiding the
+  # bug.
+
+  tests.cleanup restore
+  for rev in $(snap list snapd --all | grep disabled | awk '{print $3}'); do
+    snap remove snapd --revision="$rev"
+  done
+
+execute: |
+  if ! os.query is-pc-amd64 && ! os.query is-arm; then
+    echo "architecture not supported for this variant"
+    exit 0
+  fi
+
+  if [ "$SPREAD_REBOOT" = "0" ]; then
+    # install the special service snap which will fail to startup every other time
+    # it is started
+    snap pack test-snapd-svc-flip-flop --filename=app.snap
+
+    # install the snap so it is running
+    snap install --dangerous app.snap
+
+    # ensure that the service is active and has a running PID
+    snap services | MATCH 'test-snapd-svc-flip-flop\.svc1\s+enabled\s+active.*'
+
+    # ensure that the service was installed with Requires= which triggers the bug
+    MATCH Requires=usr-lib-snapd.mount < /etc/systemd/system/snap.test-snapd-svc-flip-flop.svc1.service
+
+    # mock the reboot command so that when we try to refresh snapd and it fails,
+    # we can observe this and reboot with spread instead
+
+    # TODO: make this a common helper for mocking expected snapd reboots, this
+    # code is shared with uc20-recovery.sh right now
+
+    # save the original systemctl command since we essentially need to mock it
+    cp /bin/systemctl /tmp/orig-systemctl
+    chmod +x /tmp/orig-systemctl
+
+    mount -o bind "$TESTSLIB/mock-shutdown" /bin/systemctl
+    tests.cleanup defer umount /bin/systemctl
+
+    # now refresh to the new version of snapd - this will trigger a reboot of 
+    # the system
+    snap install --dangerous snapd-pr.snap 2>&1 | MATCH "snapd is about to reboot the system"
+
+    # snapd schedules a slow timeout and an immediate one, however it is
+    # scheduled asynchronously, try to keep the check simple
+    # shellcheck disable=SC2016
+    retry -n 30 --wait 1 sh -c 'test "$(wc -l < /tmp/mock-shutdown.calls)" = "2"'
+    # a reboot in 10 minutes should have been scheduled
+    MATCH -- '-r \+10' < /tmp/mock-shutdown.calls
+    # and an immediate reboot should have been scheduled
+    MATCH -- '-r \+0' < /tmp/mock-shutdown.calls
+
+    # check that before shutting down, snapd rewrote the unit service file to
+    # contain Wants= instead of Requires=
+    MATCH Wants=usr-lib-snapd.mount < /etc/systemd/system/snap.test-snapd-svc-flip-flop.svc1.service
+
+    # and the service should not be running anymore
+    if [ "$(systemctl is-active snap.test-snapd-svc-flip-flop.svc.service)" != "inactive" ]; then
+      echo "the service is still active somehow, test probably broken"
+      exit 1
+    fi
+
+    # restore shutdown so that spread can reboot the host
+    tests.cleanup pop
+
+    REBOOT
+  elif [ "$SPREAD_REBOOT" = "1" ]; then
```

Comment:
I think elif is not needed, spread will execute all the lines after the if for SPREAD_REBOOT=0

Reply:
done